packages/browser/src/utils/scheduler.ts

```diff
@@ -0,0 +1,77 @@
+import { logger } from './logger'
+import { isNullish } from '@posthog/core'
+
+export type Priority = 'high' | 'normal'
+
+const DEFAULT_TIME_BUDGET_MS = 30
+
+class Scheduler {
+    private _highQueue: Array<() => void> = []
+    private _normalQueue: Array<() => void> = []
+    private _scheduled: ReturnType<typeof setTimeout> | null = null
+    private _timeBudgetMs = DEFAULT_TIME_BUDGET_MS
+
+    processEach<T, R>(items: T[], fn: (item: T, index: number) => R, options?: { priority?: Priority }): Promise<R[]> {
+        if (items.length === 0) {
+            // eslint-disable-next-line compat/compat
+            return Promise.resolve([])
+        }
+
+        const results: R[] = new Array(items.length)
+        let completedCount = 0
+        const queue = options?.priority === 'high' ? this._highQueue : this._normalQueue
+
+        // eslint-disable-next-line compat/compat
```

Comment:
I don't understand why we need the `compat/compat`​ rule now that we're polyfilling `Promise`​